packages/okta-adapter/src/weak_references/policy_rule_priorities.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ *                      Copyright 2024 Salto Labs Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  GetCustomReferencesFunc,
+  InstanceElement,
+  ReferenceInfo,
+  isInstanceElement,
+  isReferenceExpression,
+} from '@salto-io/adapter-api'
+import { collections, values } from '@salto-io/lowerdash'
+import { logger } from '@salto-io/logging'
+import { POLICY_RULE_PRIORITY_TYPE_NAMES } from '../constants'
+import { WeakReferencesHandler } from './weak_references_handler'
+
+const { awu } = collections.asynciterable
+
+const log = logger(module)
+
+const getFieldReferences = async (instance: InstanceElement): Promise<ReferenceInfo[]> => {
```

Comment:
just a suggestion, but maybe use a more descriptive name ?
```suggestion
const markRuleAsWeakReference = async (instance: InstanceElement): Promise<ReferenceInfo[]> => {
```